### PR TITLE
Fix mutation function in JoinStatusTests

### PR DESCRIPTION
**Pull request by DaveCTurner**
Sometimes `mutateJoinStatus()` would fail to return a different object, e.g. if the message was very short. Also in case 4 it was mutating both the message and the age, but should not have been mutating the message.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/coordination/JoinStatusTests.java [hunk 1]
```diff
@@ -47,16 +47,16 @@ private JoinStatus mutateJoinStatus(JoinStatus originalJoinStatus) {
...
+                    randomValueOtherThan(originalJoinStatus.message(), () -> randomAlphaOfLengthBetween(0, 30)),
                     originalJoinStatus.age()
                 );
             }
             case 3 -> {
                 return new JoinStatus(
                     originalJoinStatus.remoteNode(),
                     originalJoinStatus.term(),
-                    randomAlphaOfLength(30),
-                    randomNonNegativeTimeValue()
```
Comment (pxsalehi) [inline server/src/test/java/org/elasticsearch/cluster/coordination/JoinStatusTests.java:59]: Is the `ESTestCase` necessary here?

Comment (pxsalehi) [reply server/src/test/java/org/elasticsearch/cluster/coordination/JoinStatusTests.java:59]: hmm wrong marker, I guess!

File: server/src/test/java/org/elasticsearch/cluster/coordination/JoinStatusTests.java [hunk 2]
```diff
@@ -47,16 +47,16 @@ private JoinStatus mutateJoinStatus(JoinStatus originalJoinStatus) {
...
+                    originalJoinStatus.message(),
+                    ESTestCase.randomValueOtherThan(originalJoinStatus.age(), this::randomNonNegativeTimeValue)
```
Comment (pxsalehi) [inline server/src/test/java/org/elasticsearch/cluster/coordination/JoinStatusTests.java:59]: Is the `ESTestCase` necessary here?

**Review by pxsalehi — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/cluster/coordination/JoinStatusTests.java (hunk 2 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/coordination/JoinStatusTests.java]: No, not sure why IntelliJ put that there 😕

**Pull request merged**